arrow-buffer/src/util/bit_mask.rs

```diff
@@ -64,125 +64,235 @@ pub fn set_bits(
-        ];
-        let expected_null_count = 23;
-        let result = set_bits(
-            destination.as_mut_slice(),
-            source,
-            destination_offset,
-            source_offset,
-            len,
-        );
-
-        assert_eq!(destination, expected_data);
-        assert_eq!(result, expected_null_count);
+        SetBitsTest {
+            write_data: vec![0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+            data: vec![
+                0b11100111, 0b10100101, 0b10011001, 0b11011011, 0b11101011, 0b11000011, 0b11100111,
+                0b10100101,
+            ],
+            offset_write: 8,
+            offset_read: 0,
+            len: 62,
+            expected_data: vec![
+                0, 0b11100111, 0b10100101, 0b10011001, 0b11011011, 0b11101011, 0b11000011,
+                0b11100111, 0b00100101, 0,
+            ],
+            expected_null_count: 23,
+        }
+        .verify();
     }
 
     #[test]
     fn test_set_bits_unaligned() {
-        let mut destination: Vec<u8> = vec![0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
-        let source: &[u8] = &[
-            0b11100111, 0b10100101, 0b10011001, 0b11011011, 0b11101011, 0b11000011, 0b11100111,
-            0b10100101, 0b10011001, 0b11011011, 0b11101011, 0b11000011, 0b11100111, 0b10100101,
-            0b10011001, 0b11011011, 0b11101011, 0b11000011,
-        ];
-
-        let destination_offset = 3;
-        let source_offset = 5;
-
-        let len = 95;
-
-        let expected_data: &[u8] = &[
-            0b01111000, 0b01101001, 0b11100110, 0b11110110, 0b11111010, 0b11110000, 0b01111001,
-            0b01101001, 0b11100110, 0b11110110, 0b11111010, 0b11110000, 0b00000001,
-        ];
-        let expected_null_count = 35;
-        let result = set_bits(
-            destination.as_mut_slice(),
-            source,
-            destination_offset,
-            source_offset,
-            len,
-        );
-
-        assert_eq!(destination, expected_data);
-        assert_eq!(result, expected_null_count);
+        SetBitsTest {
+            write_data: vec![0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+            data: vec![
+                0b11100111, 0b10100101, 0b10011001, 0b11011011, 0b11101011, 0b11000011, 0b11100111,
+                0b10100101, 0b10011001, 0b11011011, 0b11101011, 0b11000011, 0b11100111, 0b10100101,
+                0b10011001, 0b11011011, 0b11101011, 0b11000011,
+            ],
+            offset_write: 3,
+            offset_read: 5,
+            len: 95,
+            expected_data: vec![
+                0b01111000, 0b01101001, 0b11100110, 0b11110110, 0b11111010, 0b11110000, 0b01111001,
+                0b01101001, 0b11100110, 0b11110110, 0b11111010, 0b11110000, 0b00000001,
+            ],
+            expected_null_count: 35,
+        }
+        .verify();
+    }
+
+    #[test]
+    fn set_bits_fuz() {
+        let mut rng = StdRng::seed_from_u64(42);
+        let mut data = SetBitsTest::new();
+        for _ in 0..10000 {
```

Comment:
10000 takes too long, lets use less

```suggestion
        for _ in 0..100 {
```